src/state/metadata/reducer.ts

```diff
@@ -50,8 +53,17 @@ export const initialState = {
 };
 
 const actionToConfigMap: TypeToDescriptionMap = {
+    [REPLACE_IMAGE_DATASET]: {
```

Comment:
I said above I think this belongs in it's own branch of state, but I also thing this action could be called `changeDataSetType` or something instead of `replace`